shell/common/rasterizer.cc

```diff
@@ -136,6 +136,13 @@ void Rasterizer::Draw(fml::RefPtr<Pipeline<flutter::LayerTree>> pipeline) {
     consume_result = PipelineConsumeResult::MoreAvailable;
   }
 
+  // Merging the thread as we know the next `Draw` should be run on the platform
+  // thread.
+  if (raster_status == RasterStatus::kResubmit) {
+    auto* external_view_embedder = surface_->GetExternalViewEmbedder();
```

Comment:
how do we know that `GetExternalViewEmbedder` doesn't return `null`?

Reply:
Only external view embedder can cause `RasterStatus::kResubmit`
I can add a DCheck to make it more explicit.